app/main.py

```diff
@@ -1,8 +1,36 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+
+        Person.people[self.name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    instances = []
+
+    # First, create all Person instances and add them to Person.people
+    for person in people:
+        name = person["name"]
+        age = person["age"]
+
+        # Create a Person instance and add it to the list
+        person_instance = Person(name, age)
+        instances.append(person_instance)
+
+    # Now set the wife/husband attributes if they are not None
+    for person in people:
+        name = person["name"]
+        spouse_key = "wife" if "wife" in person else "husband"
+
+        if person[spouse_key] is not None:
+            # Retrieve the person instance and their spouse instance
+            person_instance = Person.people[name]
+            spouse_instance = Person.people[person[spouse_key]]
```

Comment:
There is an assumption that the spouse's name will always be found in the `Person.people` dictionary. If a spouse's name is not found, this will raise a KeyError. Consider adding error handling to manage cases where the spouse is not found.